### PR TITLE
Add weakproxy Sequence, Mapping

### PR DESCRIPTION
**Pull request by oow214**

### CODE REVIEW

File: vm/src/builtins/weakproxy.rs [hunk 1]
```diff
@@ -66,6 +67,24 @@ impl PyWeakProxy {
...
     }
+
+    fn len(&self, vm: &VirtualMachine) -> PyResult<usize> {
+        match self.weak.upgrade() {
+            Some(obj) => obj.length(vm),
+            None => Err(new_reference_error(vm)),
+        }
+    }
+
+    #[pymethod(magic)]
```
Comment (Snowapril) [inline vm/src/builtins/weakproxy.rs:78]:
```suggestion
```
`weakproxy` does not have contains a python method 😊

File: vm/src/builtins/weakproxy.rs [hunk 2]
```diff
@@ -66,6 +67,24 @@ impl PyWeakProxy {
...
+    fn contains(&self, needle: PyObjectRef, vm: &VirtualMachine) -> PyResult<bool> {
+        match self.weak.upgrade() {
+            Some(obj) => {
+                // let seq = PySequence::try_protocol(obj.as_ref(), vm)?;
```
Comment (Snowapril) [inline vm/src/builtins/weakproxy.rs:82]: Maybe mistake? Is this comment intended?

**Comment by Snowapril** [comment]
Looks good! As we implement sequence protocol, `test_weakref` in `test_array.py` goes success by your changes. It calls built-in `len` which uses `length` field in sequence protocol.

File: vm/src/builtins/weakproxy.rs (hunk 1 above)
Comment (Snowapril) [inline vm/src/builtins/weakproxy.rs:78]:
```suggestion
```
`weakproxy` does not have `contains` as a python method 😊

File: vm/src/builtins/weakproxy.rs (hunk 2 above)
Comment (oow214) [reply vm/src/builtins/weakproxy.rs:82]: It's a mistake! 😅

**Comment by fanninpm** [comment]
Running this command will help you identify the tests that are unexpectedly passing:

```text
cargo run --release --features ssl,jit -- -m test.test_array -vf
```

**Review by youknowone — APPROVED** [review]
Great! Thank you for contributing! I attached a commit to refactor repeated pattern. Please check it if you are interested in.

**Pull request merged**